sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/TrigonometryTests.java

```diff
@@ -182,4 +183,924 @@ public void testPi() {
+                        "SELECT cot(0.25);\n" +
+                        " cot \n" +
+                        "-----\n" +
+                        " 3.91631736464594\n" +
+                        "(1 row)\n" +
+                        "\n" +
+                        "SELECT cot(0.5);\n" +
+                        " cot \n" +
+                        "-----\n" +
+                        " 1.830487721712452\n" +
+                        "(1 row)\n" +
+                        "\n" +
+                        "SELECT cot(0.75);\n" +
+                        " cot \n" +
+                        "-----\n" +
+                        " 1.0734261485493772\n" +
+                        "(1 row)\n" +
+                        "\n" +
+                        "SELECT cot(1);\n" +
+                        " cot \n" +
+                        "-----\n" +
+                        " 0.6420926159343306\n" +
+                        "(1 row)"
+        );
+    }
+
+    // Tested using Postgres 15.2
+    @Test
+    public void testCotDouble() {
+        this.qs(
+                        "SELECT cot(CAST(0 AS DOUBLE));\n" +
+                        " cot \n" +
+                        "-----\n" +
+                        " Infinity\n" +
+                        "(1 row)\n" +
+                        "\n" +
+                        "SELECT cot(pi);\n" +
+                        " cot \n" +
+                        "-----\n" +
+                        " -8.165619676597685e+15\n" +
+                        "(1 row)\n" +
+                        "\n" +
+                        "SELECT cot(CAST(null AS DOUBLE));\n" +
+                        " cot \n" +
+                        "-----\n" +
+                        "NULL\n" +
+                        "(1 row)\n" +
+                        "\n" +
+                        "SELECT cot(CAST(0.25 AS DOUBLE));\n" +
+                        " cot \n" +
+                        "-----\n" +
+                        " 3.91631736464594\n" +
+                        "(1 row)\n" +
+                        "\n" +
+                        "SELECT cot(CAST(0.5 AS DOUBLE));\n" +
+                        " cot \n" +
+                        "-----\n" +
+                        " 1.830487721712452\n" +
+                        "(1 row)\n" +
+                        "\n" +
+                        "SELECT cot(CAST(0.75 AS DOUBLE));\n" +
+                        " cot \n" +
+                        "-----\n" +
+                        " 1.0734261485493772\n" +
+                        "(1 row)\n" +
+                        "\n" +
+                        "SELECT cot(CAST(1 AS DOUBLE));\n" +
+                        " cot \n" +
+                        "-----\n" +
+                        " 0.6420926159343306\n" +
+                        "(1 row)"
+        );
+    }
+
+    // Tested using Postgres 15.2
+    // Postgres throws an error if the argument is out of the range: [-1, 1]
+    // Right now, we return a NaN, instead of throwing an error
```

Comment:
Maybe it's better to write that in the documentation instead of saying that it returns an "arbitrary" number.
That's important, because we want our functions to be deterministic.

Reply:
We return an arbitrary number for undefined values, ie, `COT(pi)`. 
But for functions like `asin` and `acos` if the value is outside `[-1, 1]`, we return `NaN`, this is documented.

Reply:
Ok, then say it's fixed but arbitrary. It doesn't change from run to run.

Reply:
Okay.